yt/sample_data/api.py

```diff
@@ -0,0 +1,172 @@
+"""
+This is a collection of helper functions to yt.load_sample
+"""
+import json
+import re
+from functools import lru_cache
+from itertools import chain
+from pathlib import Path
+
+import pkg_resources
+
+from yt.funcs import mylog
+from yt.utilities.on_demand_imports import (
+    _pandas as pandas,
```

Comment:
oh it's not ? Well I'll open a different PR to make it so then, because pandas is _definitely_ not a hard dependency. Not counting this line, it's only imported in 3 places in the code, 2 of which are not top level, and the third is in a try-except block.

Reply:
#3148 !

Reply:
This PR is now rebased on top of the aforementioned on.